src/main.cpp

```diff
@@ -4840,14 +4852,21 @@ bool ProcessBlock(CNode* pfrom, CBlock* pblock, bool generated_by_me)
              mi != mapOrphanBlocksByPrev.upper_bound(hashPrev);
              ++mi)
         {
-            CBlock* pblockOrphan = (*mi).second;
+            CBlock* pblockOrphan = mi->second;
             if (pblockOrphan->AcceptBlock(generated_by_me))
                 vWorkQueue.push_back(pblockOrphan->GetHash());
+            delete mapOrphanBlocks.at(pblockOrphan->GetHash());
```

Comment:
This deletion can be removed, but it should be erased from `mapOrphanBlocksByPrev`.

Reply:
Why ? wouldn't `mapOrphanBlocks.erase(pblockOrphan->GetHash());` leak memory ?

Reply:
It is deleted 3 lines later :)